### PR TITLE
feat(DNA-3821): Modify Working Timeout and Cron Schedule Options for Reports

### PR DESCRIPTION
**Pull request by SamraHanifCareem**
- [x] Set the **MAX_WORKING_TIMEOUT** validation for reports to 15 mins.
- [x] Bump `reactjs-cron` version to **2.0.0** from **1.2.0**
- [x] Adding CronSchedule allowPeriods key to dynamically populate 'Period Types'
- [x] Add validation for CronSchedule "* * * * *" i.e every minute not to be accepted for Reports.
- [x] Unit test cases for the above feature

### CODE REVIEW

File: superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx [hunk 1]
```diff
@@ -1010,7 +1028,11 @@ const AlertReportModal: FunctionComponent<AlertReportModalProps> = ({
       checkNotificationSettings()
     ) {
       if (isReport) {
-        setDisableSave(false);
+        if (currentAlert.crontab === '* * * * *') {
```
Comment (puneet-bdp) [inline superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx:1031]: Could you explain the logic behind this cron schedule?

Comment (puneet-bdp) [reply superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx:1031]: In that case we should check for the first * only as it is responsible for every minute execution. For example expression like "* 6 12 1 *" will not match the if condition above but it executes "At every minute past hour 6 on day-of-month 12 in January". Let me know if you want to discuss this in detail.

Comment (SamraHanifCareem) [reply superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx:1031]: Yea, make sense. Let me incorporate this

File: superset-frontend/src/views/CRUD/alert/AlertReportModal.test.jsx [hunk 2]
```diff
@@ -364,4 +374,69 @@ describe('AlertReportModal', () => {
...
+    };
+
+    const editReportWrapper = await mountAndWait(props);
+    expect(
+      editReportWrapper.find('[data-test="alert-report-modal-title"]').text(),
+    ).toEqual('Edit Report');
+
+    expect(
+      editReportWrapper.find('input[name="crontab"]').props().value,
+    ).toEqual('* * * * *');
```
Comment (puneet-bdp) [inline superset-frontend/src/views/CRUD/alert/AlertReportModal.test.jsx:394]: Could we add another test case for fix you recently added.

Comment (SamraHanifCareem) [reply superset-frontend/src/views/CRUD/alert/AlertReportModal.test.jsx:394]: Added

**Review by puneet-bdp — APPROVED** [review]

**Pull request merged**